server/routes/restaurant.js

```diff
@@ -32,146 +32,146 @@ router.get('/searchRestaurants/:latitude/:longitude/:radius', async (req, res) =
-                        }  
                     }
-                    //create a restaurant entry using model schema and add new entry to the RestaurantDetails collection
-                    const restaurant = new Restaurant({
-                        restaurantId: fsq_id,
-                        name: placeDetails.name,
-                        address: placeDetails.location.formatted_address,
-                        latitude: placeDetails.geocodes.main.latitude,
-                        longitude: placeDetails.geocodes.main.longitude,
-                        openingHours: placeDetails.hours.regular,
-                        phoneNumber: placeDetails.tel,
-                        website: placeDetails.website,
-                        cuisineType: cuisineType_array,
-                        restaurantType: restaurantType_array,
-                        price: placeDetails.price,
-                        rating: placeDetails.rating,
-                        total_ratings: placeDetails.stats.total_ratings,
-                        restaurantPhotos: restaurant_photo_array,
-                        foodPhotos: food_photo_array,
-                        menuId: menu_find_result[0].menuId,
-                        hasMenu: true
-                    }) 
-                   
-                    await databaseMaster.dbOp('insert', 'RestaurantDetails', { docs: [restaurant] });
-                    restaurant_ids.push(fsq_id);
-                }
-                else 
-                {
-                    //for restaurants without a valid menu, create any entry in the restaurant details collections but set the hasMenuFlag to false
-                    const restaurant = new Restaurant({
-                        restaurantId: fsq_id,
-                        hasMenu: false
-                    })
-                    await databaseMaster.dbOp('insert', 'RestaurantDetails', { docs: [restaurant] });
-                }
-            }
-            else
-            { 
-                //if the restaurant has a valid menu, add it to the list of search query restaurants
-                  if(result[0].hasMenu)
-                {
-                    restaurant_ids.push(fsq_id);
-                } 
+                    else {
+                            
+                        if(response1[0].hasMenu)
+                        {
+                                restaurant_ids.push(fsq_id);
+                        } 
+                    } 
+                });
             }
-        }  
-        //a wierd bug where the restaurant_ids array were returning duplicates, quick fix for now
-        res.json(Array.from(new Set(restaurant_ids)));
- 
+        });
+            //a wierd bug where the restaurant_ids array were returning duplicates, quick fix for now
+            res.json(Array.from(new Set(restaurant_ids)));
```

Comment:
I think we need to take a closer look at this. We've got a comment that already tells us the `restaurant_ids` array were returning duplicates, yet we only filter these out post calling the `testFlow`.

Reply:
Hey,
- I'm trying to test if the bug fix works using a duplicate MongoDB database 
- I can amend the searchRestaurants call to call create restaurant()

Reply:
Sounds good! Currently also testing with Abrar - what we've done is we use your `Set()` fix but call it prior to the for loop, fingers crossed!🤞

Reply:
Oh, I totally forgot about this bug, I think it is caused by the same error that is duplicating restaurants

Reply:
I tried to address the review changes you requested in the latest commit - added try catch {} to catch errors and reused the /createRestaurant and /getRestaurant in searchRestaurant() call